src/fragmentarium/domain/text.ts

```diff
@@ -140,21 +140,147 @@ export type Token =
   | Enclosure
   | Tabulation
 
-export type Line = RulingDollarLine | LineBase
+export type Line =
+  | TextLine
+  | EmptyLine
+  | LooseDollarLine
+  | ImageDollarLine
+  | RulingDollarLine
+  | SealDollarLine
+  | StateDollarLine
+  | SealAtLine
+  | HeadingAtLine
+  | ColumnAtLine
+  | DiscourseAtLine
+  | SurfaceAtLine
+  | ObjectAtLine
+  | DivisionAtLine
+  | CompositeAtLine
 
 export interface LineBase {
   readonly type: string
   readonly prefix: string
   readonly content: ReadonlyArray<Token>
 }
 
+export interface EmptyLine extends LineBase {
+  readonly type: 'EmptyLine'
+}
+export interface LineNumber {
+  readonly number: number
+  readonly hasPrime: false
+  readonly prefixModifier?: string
+  readonly suffixModifier?: string
+}
+export interface TextLine extends LineBase {
+  readonly type: 'TextLine'
+  readonly lineNumber?: LineNumber
+}
+
+export interface ControlLines extends LineBase {
+  readonly displayValue: string
+}
+
+export interface LooseDollarLine extends ControlLines {
+  readonly type: 'LooseDollarLine'
+  readonly text: string
+}
+
+export interface ImageDollarLine extends ControlLines {
+  readonly type: 'ImageDollarLine'
+  readonly number: string
+  readonly letter?: string | null
+  readonly text: string
+}
+
+export interface LooseDollarLine extends ControlLines {
+  readonly type: 'LooseDollarLine'
+  readonly text: string
+}
+
 export interface RulingDollarLine extends LineBase {
   readonly type: 'RulingDollarLine'
   readonly number: 'SINGLE' | 'DOUBLE' | 'TRIPLE'
-  readonly status: string | null
+  readonly status?: string | null
```

Comment:
Can status be missing?

Reply:
Why was this resolved?

Reply:
status can not be missing, I got rid of the `?` that should resolve it right ?